metamorph-clib/src/dyntypes/dyntypes.c

```diff
@@ -85,7 +88,8 @@ void release_dyntype(dyntype_t dyntype){
         case(SCHEME_TYPE_STRING): {
             scheme_string_t string;
             string = *dyntype.data.string_val;
-            //release_string(string);
+            if (dyntype._mutable)
```

Comment:
do not use `_mutable` flag of `dyntype_t` for marking strings created via c literal strings. this flag is for the scheme notion of mutable storage locations only.